### PR TITLE
HIVE-24874: Worker performance metric

### PR DESCRIPTION
**Pull request by deniskuzZ**
### What changes were proposed in this pull request?

Introduced performance metric for Compaction Worker

### Why are the changes needed?

Compaction observability

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Unit tests

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java [hunk 1]
```diff
@@ -97,12 +99,14 @@ public static String hostname() {
   public void run() {
     LOG.info("Starting Worker thread");
     boolean computeStats = conf.getBoolVar(HiveConf.ConfVars.HIVE_MR_COMPACTOR_GATHER_STATS);
+    boolean metricsEnabled = conf.getBoolVar(HiveConf.ConfVars.HIVE_SERVER2_METRICS_ENABLED);
```
Comment (pvargacl) [inline ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java:102]: I didn't see anywhere in HS2 to check this property before using perflogger, the metricsfactory will be disabled globally if this is not enabled. You can remove it.

File: ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java [hunk 2]
```diff
@@ -390,9 +392,12 @@ private ExecutorService getTimeoutHandlingExecutor() {
    * @throws InterruptedException is thrown when the process is interrupted because of timeout for example
    */
   @VisibleForTesting
-  protected Boolean findNextCompactionAndExecute(boolean computeStats) throws InterruptedException {
+  protected Boolean findNextCompactionAndExecute(boolean computeStats) {
```
Comment (pvargacl) [inline ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java:395]: please fix the javadoc also

File: ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java (hunk 1 above)
Comment (deniskuzZ) [reply ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java]: fixed

File: ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java (hunk 2 above)
Comment (deniskuzZ) [reply ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java:394]: fixed

Comment (pvargacl) [reply ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java:394]: btw InterruptedException is thrown from checkInterrupt

Comment (deniskuzZ) [reply ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java:394]: but it's handled inside of findNextCompactionAndExecute

**Review by pvargacl — APPROVED** [review]

**Pull request merged**